source/extensions/quic_listeners/quiche/platform/quic_thread_impl.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+// NOLINT(namespace-envoy)
+//
+// This file is part of the QUICHE platform implementation, and is not to be
+// consumed or referenced directly by other Envoy code. It serves purely as a
+// porting layer for QUICHE.
+
+#include <string>
+
+#include "envoy/thread/thread.h"
+
+#include "common/common/assert.h"
+
+#include "absl/synchronization/notification.h"
+
+namespace quic {
+
+// A class representing a thread of execution in QUIC.
+class QuicThreadImpl {
+public:
+  QuicThreadImpl(const std::string& /*name*/) {}
+  QuicThreadImpl(const QuicThreadImpl&) = delete;
+  QuicThreadImpl& operator=(const QuicThreadImpl&) = delete;
+
+  virtual ~QuicThreadImpl() {
+    if (thread_ != nullptr) {
+      PANIC("QuicThread should be joined before destruction.");
```

Comment:
There is a subtle problem, if the QuicThread is destructed immediately after Start(), but before the Run() function is called, then QuicThread::Run() becomes a call to a pure virtual function(since child has destructed), which aborts the program.

I can change QuicThread::Run() to a empty function to avoid the abort. How's that sound?

Reply:
I see. On second thought, allowing the QuicThreadImpl's Run() method to execute after/while the instance has been destructed seems like inviting problems. Making QuicThread::Run() an empty function feels like skating around the issue. Let's keep the PANIC as-is.

(Another option would be to do something more complicated, like having an internal enum to differentiate between the states of "not started" vs "started but Run not invoked yet" vs "Run invoked", but that feels like overkill for behavior that we shouldn't be triggering or need, anyways. I prefer what you have currently.)